packages/core/src/agent/didcomm/versions/v2/DIDCommV2Message.ts

```diff
@@ -0,0 +1,89 @@
+import type { ServiceDecorator } from '../../../../decorators/service/ServiceDecorator'
+import type { AgentMessage } from '../../../AgentMessage'
+
+import { parseUrl } from 'query-string'
+
+import { AriesFrameworkError } from '../../../../error'
+import { JsonEncoder } from '../../../../utils'
+import { JsonTransformer } from '../../../../utils/JsonTransformer'
+import { DIDCommMessageVersion } from '../../types'
+
+import { DIDCommV2BaseMessage } from './DIDCommV2BaseMessage'
+
+export class DIDCommV2Message extends DIDCommV2BaseMessage implements AgentMessage {
+  public toJSON(): Record<string, unknown> {
+    return JsonTransformer.toJSON(this)
+  }
+
+  public get version(): DIDCommMessageVersion {
+    return DIDCommMessageVersion.V2
+  }
+
+  public get threadId(): string | undefined {
+    return this.thid
+  }
```

Comment:
Can't we make this the property directly?

```suggestion
  @Expose({ name: 'thid' })
  @IsString()
  public threadId?: string
```